examples/gazebo_differential_drive_plugin/differential_drive_plugin.cc

```diff
@@ -0,0 +1,110 @@
+        return;
+      }
+
+      // Store the model pointer for convenience.
+      this->model = _model;
+
+      // Get the first 2 joint. We are making an assumption about the model
+      // having 2 joints which are the 2 wheels.
+      this->left_wheel_joint = _model->GetJoints()[1];
+      this->right_wheel_joint = _model->GetJoints()[0];
+
+      // Setup a P-controller, with a gain of 0.1.
+      this->left_wheel_pid = common::PID(10, 0, 0);
+      this->right_wheel_pid = common::PID(10, 0, 0);
+
+      // Apply the P-controller to the joint.
+      this->model->GetJointController()->SetVelocityPID(this->left_wheel_joint->GetScopedName(), this->left_wheel_pid);
+      this->model->GetJointController()->SetVelocityPID(this->right_wheel_joint->GetScopedName(), this->right_wheel_pid);
+
+      // Default to zero velocity
+      double velocity = 0;
+
+      // Check that the velocity element exists, then read the value
+      if (_sdf->HasElement("velocity"))
+        velocity = _sdf->Get<double>("velocity");
+
+      this->SetVelocity(velocity, velocity);
+
+      // Create the node
+      this->node = transport::NodePtr(new transport::Node());
+      #if GAZEBO_MAJOR_VERSION < 8
+      this->node->Init(this->model->GetWorld()->GetName());
+      #else
+      this->node->Init(this->model->GetWorld()->Name());
+      #endif
+
+      // Create a topic name
+      std::string topicName = "~/" + this->model->GetName() + "/vel_cmd";
+
+      // Subscribe to the topic, and register a callback
+      this->sub = this->node->Subscribe(topicName, &DifferentialDrivePlugin::OnMsg, this);
+    }
+
+    /// \brief Set the velocity of the Velodyne
+    /// \param[in] _vel New target velocity
+    public: void SetVelocity(const double &_left_wheel_vel, const double &_right_wheel_vel)
+    {
+      // Set the joint's target velocity.
+      this->model->GetJointController()->SetVelocityTarget(this->left_wheel_joint->GetScopedName(), -(_left_wheel_vel));
+      this->model->GetJointController()->SetVelocityTarget(this->right_wheel_joint->GetScopedName(), _right_wheel_vel);
+      // std::cerr<< this->left_wheel_joint->GetScopedName()<< ": " <<_left_wheel_vel << "." << this->right_wheel_joint->GetScopedName() << ": "<< _right_wheel_vel << std::endl;
+    }
+
+    /// \brief Handle incoming message
+    /// \param[in] _msg Repurpose a vector3 message. This function will
+    /// only use the x component.
+    private: void OnMsg(ConstVector3dPtr &_msg)
+    {
+      this->SetVelocity(_msg->x(), _msg->y());
+    }
+
+    /// \brief A node used for transport
+    private: transport::NodePtr node;
+
+    /// \brief A subscriber to a named topic.
+    private: transport::SubscriberPtr sub;
+
+    /// \brief Pointer to the model.
+    private: physics::ModelPtr model;
+
+    /// \brief Pointer to the joint.
+    private: physics::JointPtr left_wheel_joint;
+    private: physics::JointPtr right_wheel_joint;
```

Comment:
We tend to write member names as e.g. ``std::string m_MyMember``, with the ``m_`` in front and the camel case. It's our own weird thing but if you could change it that'd be great :smile: